### DISCUSSION TITLE
[✨] Switch from Eslint to Biome

### DISCUSSION THREAD

**Discussion by rogervila** [QwikDev/qwik-evolution#109] [External Library Ideas]
### Is your feature request related to a problem?

This feature is unrelated to any problem but represents a significant performance improvement for the build step since Biome linter is much faster than Eslint.

Biome provides a [migration tool](https://biomejs.dev/guides/migrate-eslint-prettier/) that makes it a breeze to migrate the Eslint configuration.

### Describe the solution you'd like

The solution would consist of a PR with the following changes:

- [ ] Migrate Eslint configuration to Biome, including rules defined in `eslint-plugin-qwik`, using Biome's [migration tool](https://biomejs.dev/guides/migrate-eslint-prettier/) 
- [ ] Install Biome and uninstall Eslint-related packages from `package.json`
- [ ] Modify the `lint` command to use Biome instead of Eslint
- [ ] Deprecate `eslint-plugin-qwik`

### Describe alternatives you've considered

AFAIK, Biome is the most performant linter on the market and, since it provides a [migration tool](https://biomejs.dev/guides/migrate-eslint-prettier/), I have not considered other alternatives.

### Additional context

Performance comparison between Eslint and Biome published on [Biome's website](https://biomejs.dev/):

![eslint-vs-biome](https://github.com/QwikDev/qwik-evolution/assets/6053012/a0258d9c-6b49-4efd-b9ef-bdc19d9b4c66)

**Comment by rogervila** [comment]
If the request gets enough support I will start the previously mentioned PR

**Comment by PatrickJS** [comment]
yes, I want to switch to Biome asap the only concerns we have are tailwind prettier plugin and qwik eslint plugin that both need to work with Biome. If there is now a path and you're willing to work with us on a PR then I don't see why not

**Comment by thejackshelton** [comment]
Hey @rogervila! This would be amazing. Biome reduces a ton of bloat and dependencies from eslint and prettier in the starters generated from the Qwik CLI, and the performance is next level.

eslint-plugin-qwik to biome is what has held us back. Would love to help out on this.

**Comment by PatrickJS** [comment]
@rogervila can you ping us in the discord so we can create a channel for this effort

**Comment by thejackshelton** [comment]
Hey @rogervila! Any status updates? Would love to help you on this effort.

**Comment by shirotech** [comment]
+1 Looking forward to this being shipped

**Comment by rogervila** [comment]
Hi @thejackshelton @shirotech I will start working on it next week together with @PatrickJS :)

**Comment by maiieul** [comment]
I'd absolutely love to use Biome instead of Eslint/prettier. But we'll need to make sure current users of `eslint-plugin-qwik` and eslint v8 aren't forced to use Biome. They might not be able to migrate if Biome isn't compatible with other tools they use.

**Comment by gioboa** [comment]
We moved this issue to `qwik-evolution` repo to create a RFC discussion for this.
Here is our [Qwik RFC process](https://github.com/QwikDev/qwik-evolution?tab=readme-ov-file#qwik-rfc-process) thanks.

**Comment by briandiephuis** [comment]
Biome is pretty awesome, but there two other fast linters in this space; shouldn't we first analyze them before switching?

- https://oxc.rs/docs/guide/usage/linter.html - OXC will probably gain a lot of interest in the coming year now that Vite will switch to it + it will probably play nice with Vite as a bundler.
- https://docs.deno.com/runtime/reference/cli/linter/ - Deno 2.0 is now Node compatible and they [claim full support for Qwik](https://youtu.be/d35SlRgVxT8?t=129).

**Reply by quiple** [reply]
OXC doesn't seem to support formatting yet.

**Reply by adaliszk** [reply]
I think what Deno meant is that Qwik has a Deno plugin, so it can run on it, though it would be nice to have a clarification.
So far, running vite and qwik directly as the main engine for development, I have been experiencing weird issues, like 1 out of 3 times when you start vite, qwik complains that it runs in an unsupported platform.

**Comment by maiieul** [comment]
@briandiephuis imo we need to provide the 3-4 as potential options. Then the question of which one should be the default on starters, etc., will depend on the state of things. If we can get Biome support when Oxc isn't ready, I'm thinking why not use Biome as a default. But when we'll get Oxc support I can definitely see it become the default.

**Comment by shairez** [comment]
Hi @rogervila and everyone following this discussion! 👋

Thank you for this proposal!

After reviewing this idea in our team meeting, we decided not to move it forward as an RFC right now. 

Supporting this feature would mean maintaining the same rules in multiple places, including in multiple languages (JS and Rust), which adds a layer of complexity. Additionally, with Oxc lint emerging as a possible alternative, we think it’s important to consider future flexibility.

We did, however, decide to promote the [proposal to disable the lint step from the dev server](https://github.com/QwikDev/qwik-evolution/issues/190), which should significantly speed up development for Qwik apps. 

For now, we’d love to see the community explore implementing this as an external solution. This approach allows us to keep focusing on getting V2 out the door asap and we do want to still support ESLint, which remains widely used, and helps maintain backward compatibility.

Thank you again for the suggestion, and we’re excited to support anyone interested in developing this as an external linter plugin! 🙌
